### ISSUE TITLE
cross compilation issue

### ISSUE THREAD

**Issue by philsuess** [ccouzens/tesseract-plumbing#6]
Hey @ccouzens !

Thanks for your tesseract crates. I'm writing an app that reads texts from cards from an older game I have lying around. I want to run my code on a Raspberry Pi and need to cross-compile your crates for `aarch64-unknown-linux-gnu`.

## Setup 

1. clone `https://github.com/ccouzens/tesseract-plumbing` 
2. `cargo install cross` 
3. add `Cross.toml`  with this content:

```
[target.aarch64-unknown-linux-gnu]
image = "ghcr.io/cross-rs/aarch64-unknown-linux-gnu:main"
pre-build = [
    "dpkg --add-architecture arm64 ",
    "apt-get update ",
    "DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends apt-utils libc6-dev:arm64 libleptonica-dev libtesseract-dev clang",
]
```
4. run `cross build --release --target=aarch64-unknown-linux-gnu` 

## Error

```
debconf: delaying package configuration, since apt-utils is not installed
   Compiling futures-util v0.3.28
   Compiling clap_derive v4.3.0
   Compiling async-reactor-trait v1.1.0
   Compiling pinky-swear v6.1.0
   Compiling tesseract-plumbing v0.10.0
error[E0432]: unresolved import `self::tesseract_sys::TessBaseAPIInit5`
 --> /cargo/registry/src/github.com-1ecc6299db9ec823/tesseract-plumbing-0.10.0/src/tess_base_api.rs:9:5
  |
9 |     TessBaseAPIInit5, TessBaseAPIMeanTextConf, TessBaseAPIRecognize, TessBaseAPISetImage,
  |     ^^^^^^^^^^^^^^^^
  |     |
  |     no `TessBaseAPIInit5` in the root
  |     help: a similar name exists in the module: `TessBaseAPIInit1`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `tesseract-plumbing` due to previous error
warning: build failed, waiting for other jobs to finish...
```

This occurs because for some reason the generated `capi_bindings.rs` only contains the methods up to `TessBaseAPIInit4` - `TessBaseAPIInit5` is missing.

## More maybe helpful details?

- The image `ghcr.io/cross-rs/aarch64-unknown-linux-gnu:main` is based on 20.04.6 LTS (Focal Fossa)
- The method in question is indeed missing in that tesseract version (4.1.1-2build2 is installed on Focal):

```
root@d30866a02c60:/tesseract-plumbing# grep -i "TessBaseAPIInit" /usr/include/tesseract/capi.h
TESS_API int TESS_CALL TessBaseAPIInit(
TESS_API int TESS_CALL TessBaseAPIInit1(TessBaseAPI* handle,
TESS_API int TESS_CALL TessBaseAPIInit2(TessBaseAPI* handle,
TESS_API int TESS_CALL TessBaseAPIInit3(TessBaseAPI* handle,
TESS_API int TESS_CALL TessBaseAPIInit4(
TESS_API int TESS_CALL TessBaseAPIInitLangMod(TessBaseAPI* handle,
TESS_API void TESS_CALL TessBaseAPIInitForAnalysePage(TessBaseAPI* handle);
TESS_API void TESS_CALL TessBaseAPIInitTruthCallback(TessBaseAPI* handle,
root@d30866a02c60:/tesseract-plumbing# 
```

Any thoughts?

**Comment by ccouzens** [comment]
Hey @philsuess ,

Your app sounds interesting. Computer vision with a Raspberry Pi in Rust sounds very cool.

It looks like `TessBaseAPIInit5` was added in [Tesseract version 5.2.0
](https://github.com/tesseract-ocr/tesseract/blob/5.2.0/src/api/capi.cpp#L245) last year.

I'm going to be away from my computer for a few days, but my ideas would be:
* See if a newer version of the tesseract library is available for your build environment.
* Or add a [feature flag](https://doc.rust-lang.org/cargo/reference/features.html) called something like `tesseract-5.2.0` and put the import and code that uses it behind the feature flag.

I should be able to approve and push a release from my phone. Otherwise I can make the change when I next get a chance.

All the best,

Chris

**Comment by philsuess** [comment]
Hey Chris,

thanks for the quick reply! I did try yesterday to get tesseract5 installed, but somehow wasn't able to.

I think your feature flag is the proper solution as tesseract4 is still wide-spread over several LTS distros.

I am not in any rush with this, but will have a look at a first implementation in the next days. Will keep you posted.

Best,

Phil

**Comment by ccouzens** [comment]
Hey Phil,

Thank you for your PR.

I've merged it and published as [0.11.0](https://crates.io/crates/tesseract-plumbing/0.11.0).
